src/Oci8/Schema/Grammars/OracleGrammar.php

```diff
@@ -168,7 +168,7 @@ public function compileTableExists()
      */
     public function compileColumnExists($table)
     {
-        return "select column_name from user_tab_columns where table_name = upper('" . $table . "')";
+        return "select column_name from all_tab_cols where table_name = upper('" . $table . "')";
```

Comment:
Last updates, I think this is how we can fix the column listing. BTW, please also fix the tests when you apply this patch. Thanks!
```php
// Grammar
    /**
     * Compile the query to determine the list of columns.
     *
     * @param $database
     * @param string $table
     * @return string
     */
    public function compileColumnExists($database, $table)
    {
        return "select column_name from all_tab_cols where  upper(owner) = upper('{$database}') and table_name = upper('{$table}')";
    }

// Builder
    /**
     * Get the column listing for a given table.
     *
     * @param  string $table
     * @return array
     */
    public function getColumnListing($table)
    {
        $database = $this->connection->getConfig('username');
        $table    = $this->connection->getTablePrefix() . $table;

        $results = $this->connection->select($this->grammar->compileColumnExists($database, $table));

        return $this->connection->getPostProcessor()->processColumnListing($results);
    }
```